### PR TITLE
fix: Add highlight when search

### PR DESCRIPTION
**Pull request by drminh2807**
# Demo

https://github.com/linagora/twake-on-matrix/assets/12546908/a4878bb4-fdc7-4df3-a388-34d774276bb9

https://github.com/linagora/twake-on-matrix/assets/12546908/0c294b9e-53b9-465a-9606-a0055d97bcc4

### CODE REVIEW

File: lib/pages/chat/chat.dart [hunk 1]
```diff
@@ -1005,7 +1005,11 @@ class ChatController extends State<Chat>
     inputFocus.requestFocus();
   }
 
-  void scrollToEventId(String eventId) async {
+  void scrollToEventIdAndHighlight(String eventId) {
+    return scrollToEventId(eventId, highlight: true);
+  }
+
+  void scrollToEventId(String eventId, {bool highlight = false}) async {
```
Comment (nqhhdev) [inline lib/pages/chat/chat.dart:1012]: `highlight` => `highlightMessage`

**Review by nqhhdev — APPROVED** [review]

File: lib/pages/chat/chat.dart [hunk 2]
```diff
@@ -1050,14 +1054,21 @@ class ChatController extends State<Chat>
...
       return;
     }
-    await scrollToIndex(eventIndex);
+    await scrollToIndex(eventIndex, highlight: highlight);
     _updateScrollController();
   }
 
-  Future scrollToIndex(int index) => scrollController.scrollToIndex(
+  Future scrollToIndex(int index, {bool highlight = false}) async {
+    await scrollController.scrollToIndex(
```
Comment (sherlockvn) [inline lib/pages/chat/chat.dart:1062]: i think `jumpToIndex` can be better for performance

Comment (sherlockvn) [reply lib/pages/chat/chat.dart:1062]: remove the long scroll animation

Comment (sherlockvn) [reply lib/pages/chat/chat.dart:1062]: or you can make it faster

**Comment by hoangdat** [comment]
@drminh2807 this color for highlight 

Screenshot 2023-11-13 at 08 18 41

File: lib/pages/chat/chat.dart (hunk 2 above)
Comment (hoangdat) [reply lib/pages/chat/chat.dart:1062]: @drminh2807 may be you forgot this comment

Comment (drminh2807) [reply lib/pages/chat/chat.dart:1062]: attempted before but it looks like UI freezed

Comment (drminh2807) [reply lib/pages/chat/chat.dart]: added loading, please see demo

**Review by sherlockvn — APPROVED** [review]

**Pull request merged**